src/components/CountryList.js

```diff
@@ -0,0 +1,103 @@
+import { Link } from 'react-router-dom';
+import { BsArrowRightCircle } from 'react-icons/bs';
+import ReactPaginate from 'react-paginate';
+import { fetchApi } from '../Redux/population/populationSlice';
+import Country from './Country';
+import styles from './country.module.css';
+
+const CountryList = () => {
+  const countryList = useSelector((state) => state.populationSlice.countryList);
+
+  const dispatch = useDispatch();
+
+  const [search, setSearch] = useState('');
+
+  const [currentPage, setCurrentPage] = useState(0);
+
+  const isMobile = window.innerWidth <= 768;
+  const itemsPerPage = isMobile ? 4 : 8;
+
+  const handleChange = (e) => {
+    setSearch(e.target.value.toLowerCase());
+    setCurrentPage(0);
+  };
+
+  useEffect(() => {
+    if (!countryList) {
+      dispatch(fetchApi());
+    }
+  }, [dispatch, countryList]);
+
+  const handlePageChange = (selected) => {
+    setCurrentPage(selected.selected);
+  };
+
+  const filteredItems = countryList
+    ? countryList.filter((filtered) => {
+      if (search === '') {
+        return filtered;
+      }
+      return filtered.name.toLowerCase().includes(search);
+    })
+    : [];
+
+  const pageCount = Math.ceil(filteredItems.length / itemsPerPage);
+  const displayedItems = filteredItems.slice(
+    currentPage * itemsPerPage,
+    (currentPage + 1) * itemsPerPage,
+  );
+
+  return (
+    <>
+      <input
+        className="search-input"
+        type="text"
+        value={search}
+        aria-label="search"
+        onChange={handleChange}
+        placeholder="Search a country..."
+      />
+      <div className={styles.countryContainer}>
+
+        <ul className={styles.countrylist}>
+          {displayedItems.length > 0 ? (
+            displayedItems.map((country) => (
+              <li key={country.numericCode} className={styles.list}>
+                <Link to={`/${country.numericCode}`} className={styles.singleCountry}>
+                  <BsArrowRightCircle className={styles.direct} />
+                  <Country country={country} detailed />
+                </Link>
+              </li>
+            ))
+          ) : (
+            <p className="max-width flex-center">No matching countries found.</p>
+          )}
+        </ul>
+        {/* Pagination component */}
+        {countryList && (
+        <ReactPaginate
+          previousLabel="<"
+          nextLabel=">"
+          breakLabel=""
+          pageCount={pageCount}
+          pageRangeDisplayed={3}
+          marginPagesDisplayed={0}
+          onPageChange={handlePageChange}
+          className={styles.pagination}
+          pageClassName={styles.pageItem}
+          pageLinkClassName={styles.pageLink}
+          previousClassName={styles.pageItem}
+          previousLinkClassName={styles.pageLink}
+          nextClassName={styles.pageItem}
+          nextLinkClassName={styles.pageLink}
+          activeClassName={styles.active}
+        />
+        )}
+      </div>
+    </>
+  );
+};
```

Comment:
- [ ] You have done great work on this project 👏. In terms of design, there is a minor detail that you left out. If you look at the home page of the design sample given, it has a headline section that contains an image. Your design is missing the headline section. Kindly include it in the design of your Home Page. 👍 💻 

| Your Design | Design Template |
| ------------- | -------------------|
| ![image](https://github.com/deniodev/Countries-Population-Metrics/assets/77106962/6da8eca4-0c80-48af-9dd5-40629a689e0f) | ![image](https://github.com/deniodev/Countries-Population-Metrics/assets/77106962/69c09638-7669-4368-84cd-30a5641c9403) |

#### Hint: In my opinion, you can use a transparent image of the world but it is up to you to choose what suits your user story. 😎